### DISCUSSION TITLE
How to configure username and password while using spring integration

### DISCUSSION THREAD

**Discussion by farhan-ejaz** [eclipse-rdf4j/rdf4j#4226] [Q&A]
Any sample on how to set username password to RemoteRepositoryManager while using the spring integration

Currently, there is no property to configure which is really weird. Authentication is an important aspect and most remote rdf stores have authentication

**Comment by hmottestad** [comment]
Does this help? https://rdf4j.org/documentation/tools/server-workbench/#access-rights-and-security

**Reply by farhan-ejaz** [reply]
This is for configuring authentication on the server. My need is how to do it for the rdf4j client (while using the rdf4j-spring as a client library to connect to an rdf4 server)
![image](https://user-images.githubusercontent.com/19402644/194264878-79207e98-f67d-4a27-84ef-9b35bff2eae5.png)

The spring library provides no way to set this

**Comment by patrickwyler** [comment]
My pullrequest #4239 will solve this issue.

Added  two new properties:

```
 rdf4j.spring.repository.remote.username=admin
 rdf4j.spring.repository.remote.password=1234
```
